envi/archs/arm/disasm.py

```diff
@@ -220,107 +229,123 @@ def p_dp_imm_shift(opval, va):
     return (opcode, dp_mnem[ocode], olist, iflags)
 
 # specialized mnemonics for p_misc
-qop_mnem = ('qadd','qsub','qdadd','qdsub')
-smla_mnem = ('smlabb','smlabt','smlatb','smlatt',)
-smlal_mnem = ('smlalbb','smlalbt','smlaltb','smlaltt',)
+qop_mnem = ('qadd','qsub','qdadd','qdsub') # used in misc1
+smla_mnem = ('smlabb','smlatb','smlabt','smlatt',)
+smlal_mnem = ('smlalbb','smlaltb','smlalbt','smlaltt',)
 smul_mnem = ('smulbb','smultb','smulbt','smultt',)
 smlaw_mnem = ('smlawb','smlawt',)
 smulw_mnem = ('smulwb','smulwt',)
 
 def p_misc(opval, va):  
     # 0x0f900000 = 0x01000000 or 0x01000010 (misc and misc1 are both parsed at the same time.  see the footnote [2] on dp instructions in the Atmel AT91SAM7 docs
-    if   opval & 0x0fc00000 == 0x01000000:
-        opcode = (IENC_MISC << 16) + 1
-        mnem = 'mrs'
-        r = (opval>>22) & 1
-        Rd = (opval>>12) & 0xf
-        olist = (
-            ArmRegOper(Rd, va=va),
-            ArmPgmStatRegOper(r),
-        )
-    elif opval & 0x0fb002f0 == 0x01200000:
+
```

Comment:
in the future, when rearranging code-blocks, please do that in a separate commit.  it helps when trying to view the diffs and see what changed.